### ISSUE TITLE
Tensorboard multiple scalar summaries in one plot

### ISSUE THREAD

**Issue by abred** [tensorflow/tensorflow#7089]
In 0.12 I could do this:
> testAccuracy = tf.Variable(0., name="accuracy")
trainAccuracy = tf.Variable(0., name="accuracy")
testAccSum = tf.scalar_summary("Accuracy", testAccuracy)
trainAccSum = tf.scalar_summary("Accuracy", trainAccuracy)
summaryOpsTest = tf.merge_summary([testAccSum])
summaryOpsTrain = tf.merge_summary([trainAccSum])
writerTest = tf.train.SummaryWriter(out_dir+'/test', sess.graph)
writerTrain = tf.train.SummaryWriter(out_dir+'/train', sess.graph)

to view both scalar summaries in a single plot.

In 0.12.1 I changed it to this:
> testAccuracy = tf.Variable(0., name="accuracy") 
trainAccuracy = tf.Variable(0., name="accuracy")
testAccSum = tf.summary.scalar("Accuracy", testAccuracy)
trainAccSum = tf.summary.scalar("Accuracy", trainAccuracy)
summaryOpsTest = tf.summary.merge([testAccSum])
summaryOpsTrain = tf.summary.merge([trainAccSum])
writerTest = tf.summary.FileWriter(out_dir+'/test', sess.graph)
writerTrain = tf.summary.FileWriter(out_dir+'/train', sess.graph)

due to the issued warnings.
But now each summary is drawn in its own plot (and the name is made unique (+"_1"))

Is this behavior intended? If so, is there a new way to achieve two summaries in one plot?

Thanks

**Comment by drpngx** [comment]
In general this is a question that is best answered on stackoverflow, where we monitor all questions with the tag `tensorflow`. What is the warning you are getting?

@dandelionmane may know how to plot two scalars in the same plot.

**Comment by abred** [comment]
yeah, thought about that, but I posted it here because I thought it might be a bug as it was working before the update.
(just the warnings that scalar_summary, merge_summary etc are deprecated)

**Comment by drpngx** [comment]
@dandelionmane might know the answer.

**Comment by priyankamandikal** [comment]
@abred , what's testAccSumDiff here?

**Comment by abred** [comment]
@priyankamandikal ah sorry, just some other summary, forgot to remove it, edited the first post

**Comment by dandelionmane** [comment]
Sorry, you can no longer use the summary ops to put both summaries in the same plot. 
If you want to work around this limitation, you can generate the Summary protocol buffers yourself, and then manually add them to the summary.FileWriter. But it's cumbersome.

(Why do you have two sequences which are different nodes, but should be displayed in the same plot? Why not run test and train data through the same subsections of the graph?)

**Comment by abred** [comment]
ok.

Well, I do run test and train data through the same part of the graph. Not sure what you're getting at, maybe I'm on the wrong track here, but I want to calculate the accuracy for training batches and for the test set und plot these, I can't use a single scalar summary for that right? What I want is something like this: http://cs231n.github.io/assets/nn3/accuracies.jpeg

**Comment by lemairecarl** [comment]
I want exactly the same as @abred. With Tensorflow 0.11, I produced an image summary with matplotlib. At each epoch the image was overwritten.
Now, with Tensorflow 1.1, the plots are kept for every epochs! I only need the plot for the last epoch.

Plotting the training error and the test error in the same plot is a very common thing to do! Do people using Tensorboard give up on that?

**Comment by omasoud** [comment]
You can do this with by creating a different FileWriter (that writes to a different log) for each curve and using the same name/tag for the things you write.

**Comment by abred** [comment]
thanks for the suggestion, but as I wrote, this does not work any more since rc0.12.
Every name in a graph is made unique by tensorflow, so if you create multiple summaries with the same name you end up with "_X" suffixes. It worked before because summaries used to use tags and not the names.

**Comment by BarzinM** [comment]
@abred and @lemairecarl , Here is what worked for me. I used one variable with two writers:
```python
import tensorflow as tf
from numpy import random

writer_1 = tf.summary.FileWriter("./logs/plot_1")
writer_2 = tf.summary.FileWriter("./logs/plot_2")

log_var = tf.Variable(0.0)
tf.summary.scalar("loss", log_var)

write_op = tf.summary.merge_all()

session = tf.InteractiveSession()
session.run(tf.global_variables_initializer())

for i in range(100):
    # for writer 1
    summary = session.run(write_op, {log_var: random.rand()})
    writer_1.add_summary(summary, i)
    writer_1.flush()

    # for writer 2
    summary = session.run(write_op, {log_var: random.rand()})
    writer_2.add_summary(summary, i)
    writer_2.flush()

```
![image](https://cloud.githubusercontent.com/assets/5580724/25247459/d9c36764-25d8-11e7-90ba-3775a84ab079.png)

**Comment by abred** [comment]
ohh, damn. well, now that you're saying it it's kind of obvious, one variable - two writers not two variables - two writers.
thanks ! 👍

**Comment by AbhinavDS** [comment]
@abred , were you able to get it working?  How did you cater to the "_X" suffixes problem ?

**Comment by abred** [comment]
@AbhinavDS: @BarzinM's solution worked for me

**Comment by simaoh** [comment]
@dandelionmane @BarzinM,  Can we do this inside tensorflow's  tf.contrib.learn.Estimator?
Contrib.learn abstracts graph, and session, so we cannot call 
```python
    summary = session.run(write_op, {log_var: random.rand()})
    writer_1.add_summary(summary, i)
```

I have been trying to use  tf.train.SummarySaverHook without success.
As an example how can I plot loss and loss2 in the same graph for the example below"

```python
def cnn_model_fn(features, labels,mode, params):
    
    LEARNING_RATE = params['learning_rate']
    
    #input
    
    input_layer=tf.reshape(features['x'],shape=[-1,28,28,1])

    #conv1
    #with tf.variable_scope("conv2d"):
    conv1=tf.layers.conv2d(inputs=input_layer,
                           filters=32,
                           kernel_size=[5, 5],
                           padding='same',
                           activation=tf.nn.relu)
    
    pool1=tf.layers.max_pooling2d(inputs=conv1, pool_size=[2,2], strides=2)

    #conv2
    conv2=tf.layers.conv2d(inputs=pool1,
                           filters=64,
                           kernel_size=[5,5],
                           padding='same',
                           activation=tf.nn.relu, name='my_conv2')
    pool2=tf.layers.max_pooling2d(inputs=conv2, pool_size=[2,2], strides=2)

    #fc1
    flat=tf.reshape(pool2, [-1, 7*7*64])
    
    batch_norm=tf.contrib.layers.batch_norm(flat, scale=True, scope='batch_norm')
    #trainable
    dense1=tf.layers.dense(batch_norm, 1024, activation=tf.nn.relu)
    
    dropout = tf.layers.dropout(inputs=dense1, rate=0.4, training=mode == learn.ModeKeys.TRAIN)

    #fc2
    logits=tf.layers.dense(dropout, 10, activation=tf.nn.relu)
    loss = None
    train_op = None
    
    
    #loss
    if mode != learn.ModeKeys.INFER:
        onehot_labels=tf.one_hot(indices=tf.cast(labels, tf.int32),depth=10)
        loss=tf.losses.softmax_cross_entropy(onehot_labels=onehot_labels, logits=logits)
        loss2= tf.losses.sigmoid_cross_entropy(multi_class_labels=onehot_labels, logits=logits)
    
    #optimizer
    if mode == learn.ModeKeys.TRAIN:
        train_op = tf.contrib.layers.optimize_loss(
                        loss=loss,
                        global_step=tf.contrib.framework.get_global_step(),
                        learning_rate=LEARNING_RATE,
                        optimizer="Adam")
        
    predictions={
            'classes': tf.argmax(logits, axis=1) ,
            'predictions': tf.nn.softmax(logits,name="softmax_tensor")           
    }
    
    return model_fn_lib.ModelFnOps(mode=mode, predictions=predictions, loss=loss, train_op=train_op)


original_model_dir="/tmp/bnorm/"

#Instantiate an estimator
LEARNING_RATE = 0.0001
my_params = {'learning_rate':LEARNING_RATE}

if os.path.isdir(original_model_dir):
    shutil.rmtree(original_model_dir)
runConfig = learn.RunConfig(save_summary_steps=20, save_checkpoints_steps=100, 
                            save_checkpoints_secs=None)
classifier=learn.Estimator(model_fn=cnn_model_fn, model_dir=original_model_dir,config=runConfig, params= my_params)

# Configure the accuracy metric for evaluation
metrics = {
  "accuracy":
      learn.MetricSpec(
          metric_fn=tf.metrics.accuracy, prediction_key="classes"),
}

experiment=learn.Experiment(estimator=classifier, 
                train_input_fn=train_input_fn,
                eval_input_fn=eval_input_fn,
                eval_metrics=metrics,
                train_steps=200)
experiment.train_and_evaluate()
```

**Comment by micklexqg** [comment]
@BarzinM , great demo, thank you. I have one question. if with two writers, then there should be two events(and under different path in your code). how to start tensorboard for the two writer?

**Comment by anthnyprschka** [comment]
@micklexqg You should be able to launch tensorboard with --logdir being the directory that contains both of your events files

**Comment by micklexqg** [comment]
@anthnyprschka , thanks!

**Comment by david-ngo-sixense** [comment]
Hello,

Do you know how to plot multiple scalar on the same plot using tensorflow 2.3?
For example, I want to display train_accuracy and valid_accuracy on the same plot, and train_loss and valid_loss on the same plot.
Thank you.
